File: Data Structures and Algorithms/4. Recursion/Kunal/Level 1/_8_leetcode_number_of_steps_to_reduce_a_number_to_zero.java
import java.util.*;

public class _8_leetcode_number_of_steps_to_reduce_a_number_to_zero {
    public static int number_of_steps(int n) {
        return helper(n, 0);
    }
    private static int helper(int n, int steps) {
        if(n == 0) {
            return steps;
        }
        if(n % 2 == 0) {
            return helper(n / 2, steps + 1);
        }
        return helper(n - 1, steps + 1);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int ans = number_of_steps(n);
        System.out.printf("Number of steps are: %d\n", number_of_steps(n));
    }
}

File: Data Structures and Algorithms/4. Recursion/Kunal/Level 1/_8_leetcode_number_of_steps_to_reduce_a_number_to_zero.cpp
#include <iostream>
using namespace std;

int helper(int n, int c) {
    if(n == 0)
        return c;
    if(n % 2 == 0) {
        return helper(n / 2, c + 1);
    }
    return helper(n - 1, c + 1);
}

int steps(int n) {
    return helper(n, 0);
}

int main() {
    int n;
    cin >> n;

    int ans = steps(n);
    cout << "Number of steps are: ";
    cout << ans << '\n';
    return 0;
}